src/protocol/attribution/input.rs

```diff
@@ -238,15 +247,15 @@ impl<F: Field, T: Arithmetic<F>> Resharable<F> for MCAccumulateCreditInputRow<F,
 
         let (breakdown_key, mut fields) = try_join(
             f_breakdown_key,
-            try_join_all([f_trigger_bit, f_helper_bit, f_value]),
+            try_join_all([f_value, f_helper_bit, f_trigger_bit]),
         )
         .await?;
 
         Ok(MCAccumulateCreditInputRow {
             breakdown_key,
-            is_trigger_report: fields.remove(0),
-            helper_bit: fields.remove(0),
-            trigger_value: fields.remove(0),
+            is_trigger_report: fields.pop().unwrap(),
+            helper_bit: fields.pop().unwrap(),
+            trigger_value: fields.pop().unwrap(),
```

Comment:
Rather than collecting these into a Vec, you should probably change to using a tuple.  That will (might?) avoid a heap allocation for the Vec and you can use a destructuring assignment, like:

```rust
        let (breakdown_key, (value, helper_bit, trigger_bit)) = try_join(
            f_breakdown_key,
            try_join3(f_value, f_helper_bit, f_trigger_bit),
        )
        .await?;
```

That also avoids the unnatural reverse order you have here.

Reply:
Thanks for the suggestion!